float64[] control
float64 duration